src/olympia/devhub/views.py

```diff
@@ -1837,3 +1842,101 @@ def logout(request):
     logout_user(request, response)
 
     return response
+
+
+def get_promoted_subscription_or_404(addon):
+    if not waffle.switch_is_active('enable-subscription'):
+        log.info(
+            'cannot retrieve a promoted subscription because waffle switch '
+            'is disabled.'
+        )
+        raise http.Http404()
+
+    qs = PromotedSubscription.objects.select_related('promoted_addon')
+    return get_object_or_404(qs, promoted_addon__addon=addon)
+
+
+@dev_required
+@csp_update(
+    SCRIPT_SRC="https://js.stripe.com",
+    CONNECT_SRC="https://api.stripe.com",
+    FRAME_SRC=["https://js.stripe.com", "https://hooks.stripe.com"],
+)
+def onboarding_subscription(request, addon_id, addon):
+    sub = get_promoted_subscription_or_404(addon=addon)
+
+    fields_to_update = {}
+    if addon.has_author(request.user) and not sub.link_visited_at:
```

Comment:
`@dev_required` should ensure `has_author()` already. But I'm thinking we probably want to check that the user is an owner and not just a developer ?

Reply:
I use `has_author()` to only set the `link_visited_at` when the person who is supposed to open the link has opened it. That way, admins can browse the page too.